### PR TITLE
feat: auth modal tracking

### PR DESCRIPTION
**Pull request by capJavert**
## Changes

- add tracking pixel hook for signup to work on `AuthModal` (and not only `/onboarding`

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://auth-modal-tracking.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/components/auth/AuthModal.tsx [hunk 1]
```diff
@@ -45,6 +46,12 @@ export default function AuthModal({
     closeLogin();
   };
 
+  const onSuccessfulRegistration = (newUser?: LoggedUser | AnonymousUser) => {
+    loginState?.onRegistrationSuccess?.(newUser);
+
+    logSignUp({ experienceLevel: (newUser as LoggedUser)?.experienceLevel });
```
Comment (capJavert) [inline packages/shared/src/components/auth/AuthModal.tsx:52]: I added it globally to all calls but technically inside `logSignUp` method there is a check if pixel tracking is intialized so for example it won't fire from post page. I think thats ok @idoshamun @rebelchris ?

File: packages/shared/src/components/profile/SquadsList.tsx [hunk 2]
```diff
@@ -60,8 +60,8 @@ function SquadItem({
       showLogin({
         trigger: AuthTriggers.JoinSquad,
         options: {
-          onLoginSuccess: joinSquad,
-          onRegistrationSuccess: joinSquad,
+          onLoginSuccess: () => joinSquad(),
```
Comment (capJavert) [inline packages/shared/src/components/profile/SquadsList.tsx:63]: Function was always sending `(user) => unknown` but now TS was complaining when I updated correct type in `AuthModal`

Comment (nensidosari) [reply packages/shared/src/components/profile/SquadsList.tsx:63]: Do we need the return anywhere? if not, maybe instead we can remove the return

Comment (capJavert) [reply packages/shared/src/components/profile/SquadsList.tsx:63]: Wanted to keep it as is that if we need to return we don't have to change all implementations again. Its not really doing anything.

**Review by rebelchris — APPROVED** [review]
Looks ok to me, but let's wait for @idoshamun final say if it's ok

**Review by idoshamun — APPROVED** [review]

**Pull request merged**